### PR TITLE
Tools: fix Host Exerciser Continuous mode hang

### PR DESCRIPTION
**Pull request by anandaravuri**
- Added command line Continuous mode execution time in seconds,
   Host Exerciser executes Continuous mode in command line input number of seconds

  - Exerciser executes Continuous mode  infinite loop if user doesn’t specify Continuous mode execution time,
     press CTL+C to exit HE sample.

  - Set Force test complete bit after Continuous mode  time lapse or user press CTL+C to gracefully  exit

   - Removed --perf command line input.

sudo ./bin/host_exerciser lpbk
[2021-08-25 16:53:03.391] [lpbk] [info] starting test run, count of 1
Input Config:0
Allocate SRC Buffer
Allocate DST Buffer
Allocate DSM Buffer
Start Test
Host Exerciser swtest msg:0

Host Exerciser Performance Counter:
Host Exerciser numReads:1024
Host Exerciser numWrites:1025
Host Exerciser numPendReads:0
Host Exerciser numPendWrites:0
Number of clocks:14231
Total number of Reads sent:1024
Total number of Writes sent :1022
Bandwidth: 1.15 GB/s
[2021-08-25 16:53:03.393] [lpbk] [info] Test lpbk(1): PASS

sudo ./bin/host_exerciser   --continuousmode true --contmodetime 10 lpbk
[2021-08-25 16:53:39.007] [lpbk] [info] starting test run, count of 1
Input Config:2
Allocate SRC Buffer
Allocate DST Buffer
Allocate DSM Buffer
Start Test
host exerciser continuous mode time:10 seconds
Ctrl+C  to stop continuous mode
[2021-08-25 16:53:53.013] [lpbk] [info] Test lpbk(1): PASS

 sudo ./bin/host_exerciser   --continuousmode true --contmodetime 30 lpbk
[2021-08-25 16:54:30.447] [lpbk] [info] starting test run, count of 1
Input Config:2
Allocate SRC Buffer
Allocate DST Buffer
Allocate DSM Buffer
Start Test
host exerciser continuous mode time:30 seconds
Ctrl+C  to stop continuous mode
^CHE signal handeler exit app
[2021-08-25 16:54:35.450] [lpbk] [info] Test lpbk(1): PASS

Signed-off-by: anandaravuri <ananda.ravuri@intel.com>

### CODE REVIEW

File: samples/host_exerciser/host_exerciser_cmd.h [hunk 1]
```diff
@@ -34,6 +34,16 @@ using test_afu = opae::afu_test::afu;
...
 using opae::fpga::types::token;
 
+// HE exit global flag
+volatile bool g_he_exit = false;
+
+// host exerciser signal handler
+void he_sig_handler(int)
+{
+    g_he_exit = true;
+    printf("HE signal handeler exit app \n");
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply samples/host_exerciser/host_exerciser_cmd.h]: fixed typo

File: samples/host_exerciser/host_exerciser_cmd.h [hunk 2]
```diff
@@ -91,26 +101,105 @@ class host_exerciser_cmd : public test_command
...
+        /* Wait for test completion */
+        uint32_t           timeout = HELPBK_TEST_TIMEOUT;
+        volatile uint8_t* status_ptr = dsm_->c_type();
+
+        while (0 == ((*status_ptr) & 0x1)) {
+            usleep(HELPBK_TEST_SLEEP_INVL);
+            if (--timeout == 0) {
+            std::cout << "HE LPBK TIME OUT" << std::endl;
+            host_exerciser_errors();
+            return false;
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply samples/host_exerciser/host_exerciser_cmd.h]: fixed

File: samples/host_exerciser/host_exerciser_cmd.h [hunk 3]
```diff
@@ -91,26 +101,105 @@ class host_exerciser_cmd : public test_command
...
+            std::cout << "host exerciser continuous mode time:"<<
+	            host_exe_->he_contmodetime_ << " seconds" << std::endl;
+            std::cout << "Ctrl+C  to stop continuous mode" << std::endl;
+            while (!g_he_exit) {
+                sleep(1);
+                count++;
+                if (count > host_exe_->he_contmodetime_)
+                    break;
+            }
+        he_foretestcmpl();
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply samples/host_exerciser/host_exerciser_cmd.h]: fixed

File: samples/host_exerciser/host_exerciser_cmd.h [hunk 4]
```diff
@@ -91,26 +101,105 @@ class host_exerciser_cmd : public test_command
...
+            if (--timeout == 0) {
+            std::cout << "HE LPBK TIME OUT" << std::endl;
+            host_exerciser_errors();
+            return false;
+            }
+        }
+        return true;
+    }
+
+    void he_foretestcmpl()
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply samples/host_exerciser/host_exerciser_cmd.h]: fixed typo

File: samples/host_exerciser/host_exerciser.cpp [hunk 5]
```diff
@@ -25,16 +25,24 @@
...
+void he_sig_handler(int);
 
 int main(int argc, char* argv[])
 {
   host_exerciser::host_exerciser app;
   app.register_command<host_exerciser::host_exerciser_lpbk>();
   app.register_command<host_exerciser::host_exerciser_mem>();
+  
+  // host exerciser signal handler
+  struct sigaction act_old, act_new;
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply samples/host_exerciser/host_exerciser.cpp:43]: fixed memset to 0

**Review by fpgamatt — APPROVED** [review]

**Review by tswhison — APPROVED** [review]

**Pull request merged**